apiserver/tools.go

```diff
@@ -233,61 +246,74 @@ func (h *toolsDownloadHandler) getToolsForRequest(r *http.Request, st *state.Sta
 func (h *toolsDownloadHandler) fetchAndCacheTools(
 	v version.Binary,
 	storageVers version.Binary,
-	stor binarystorage.Storage,
 	st *state.State,
-) (binarystorage.Metadata, io.ReadCloser, error) {
+) error {
 	md := binarystorage.Metadata{Version: v.String()}
 
-	model, err := st.Model()
+	controllerModel, err := st.ControllerModel()
```

Comment:
Another way of doing this is to pass in pool.SystemState() and then st.Model() - system state is the controller model state.
This also means you don't need a special ControllerToolsStorage() method either